server/src/main/java/org/opensearch/indices/replication/SegmentReplicationTargetService.java

```diff
@@ -57,43 +57,16 @@ public class SegmentReplicationTargetService implements IndexEventListener {
     private final ThreadPool threadPool;
     private final RecoverySettings recoverySettings;
 
-    private final ReplicationCollection<SegmentReplicationTarget> onGoingReplications;
+    protected final ReplicationCollection<SegmentReplicationTarget> onGoingReplications;
```

Comment:
Can we defer relaxing this scope until there is a need to.

Reply:
Done, overloaded the constructor here to pass this in for unit tests instead.